### PR TITLE
Fixes #15543: Implemented ElasticSearch Index Template Ingestion

### LINKED ISSUE #15543: Template for elasticsearch
Hello. Is it possible to group indexes by templates when working with ElasticSearch data assets? F.e. There are indexes for every day with the same package of fields. So I would like not to describe it day by day again. I see that it is possible to get index templates in https://datahubproject.io/docs/generated/ingestion/sources/elasticsearch (ingest_index_templates: True) imageWe want to see an implementation similar to the datahub implementation. This would be very convenient. We talk about this https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html 
And this is a widely used case in ELK

### PR DESCRIPTION
**Pull request by keshavmohta09**
### Describe your changes:

Fixes  #15543 

I worked on Elastic Search index template ingestion

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [x] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/search/elasticsearch/metadata.py [hunk 1]
```diff
@@ -143,6 +145,48 @@ def yield_search_index_sample_data(
                 )
             )
 
+    def get_search_index_template_list(self) -> Iterable[dict]:
+        """
+        Get List of all search index template
+        """
+        yield from self.client.indices.get_index_template()["index_templates"]
```
Comment (ulixius9) [inline ingestion/src/metadata/ingestion/source/search/elasticsearch/metadata.py:152]:
```suggestion
        yield from self.client.indices.get_index_template().get("index_templates",[])
```

File: openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json [hunk 2]
```diff
@@ -256,8 +262,20 @@
       "type": "string",
       "minLength": 1,
       "maxLength": 32
+    },
+    "indexType": {
+      "description": "Whether the entity is index or index template.",
+      "type": "string",
+      "javaType": "org.openmetadata.schema.entity.type.indexType",
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json:269]: "javaType": "org.openmetadata.schema.entity.type.SearchIndexType",
```suggestion
      "javaType": "org.openmetadata.schema.entity.type.SearchIndexType",
```

File: openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json [hunk 3]
```diff
@@ -256,8 +262,20 @@
...
+      "enum": ["Index", "Index Template"],
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json:270]:
```suggestion
      "enum": ["Index", "IndexTemplate"],
```

File: openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json (hunk 2 above)
Comment (keshavmohta09) [reply openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json:269]: done

File: openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json (hunk 3 above)
Comment (keshavmohta09) [reply openmetadata-spec/src/main/resources/json/schema/entity/data/searchIndex.json:270]: done

**Review by ulixius9 — APPROVED** [review]

**Pull request merged**